### PR TITLE
Rename AlchemicalFactory to AbsoluteAlchemicalFactory

### LINKED ISSUE #204: Rename AlchemicalFactory to AbsoluteAlchemicalFactory
We'll soon have a `RelativeAlchemicalFactory` or `HybridAlchemicalFactory` too, so it would be useful to make this change sooner rather than later.

### PR DESCRIPTION
**Pull request by andrrizzi**
Fix #204.

@jchodera is this what you had in mind for the API warning (second commit)?

### CODE REVIEW

File: openmmtools/alchemy.py [hunk 1]
```diff
@@ -889,6 +889,10 @@ def get_energy_components(cls, alchemical_system, alchemical_state, positions, p
     def replace_reaction_field(self, reference_system):
         """Replace reaction-field electrostatics with Custom*Force terms to ensure c_rf = 0.
 
+        .. warning:: Unstable API.
+            This method is still experimental. It could be moved in some
```
Comment (jchodera) [inline openmmtools/alchemy.py]: Change "moved in" to "moved to"

**Comment by jchodera** [comment]
Yes!

Maybe we should use this opportunity to change `alchemy` into a subpackage as well?

**Comment by andrrizzi** [comment]
> Change "moved in" to "moved to"

Will do! Thanks for catching.

> Maybe we should use this opportunity to change `alchemy` into a subpackage as well?

Can we postpone this until we'll incorporate the other factories? I'd like to merge everything/release everything and restart the calculations before the weekend.

**Comment by jchodera** [comment]
Sure, looks good otherwise.

**Comment by andrrizzi** [comment]
Thanks! Merging.

**Pull request merged**